robot_smach_states/src/robot_smach_states/manipulation/manipulation.py

```diff
@@ -55,6 +39,53 @@ def execute(self, userdata=None):
         return "failed"
 
 
+class HandOver(smach.State):
+    def __init__(self, robot, arm_designator, timeout=10):
+        """
+        Handover the object in arm to a human.
+        :param robot: robot to execute state with
+        :param arm_designator: designator that resolves to arm holding an object
+        :param timeout: float time the operation may take
+        """
+        smach.State.__init__(self, outcomes=['succeeded', 'failed'])
+
+        self.robot = robot
+        check_type(arm_designator, PublicArm)
+        self.arm_designator = arm_designator
+        self.timeout = timeout
+
+    def execute(self, userdata=None):
+        arm = self.arm_designator.resolve()
+        if not arm:
+            rospy.logerr("Could not resolve arm")
+            return "failed"
+
+        object_in_arm = arm.occupied_by
+        if not object_in_arm:
+            rospy.logerr("Unoccupied arm sent to HandOverToHuman, arm should be occupied when entering this state.")
```

Comment:
What should happen if the arm is unoccupied? If it returns failed we save 10 seconds but it is possible that the robot is mistaken in whether or not it is carrying something.